# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RDDProcessImpl.cpp
# Commit: dd05584049c6
# Full Hash: dd05584049c6e6138dd7dfe6067ebd03e1219782
# Author: Michael Froman <mfroman@mozilla.com>
# Date: 2019-04-10 09:58:56
# Regressor Bug: 1539029
# File Overlap Count: 1
# Description:
#   Bug 1539029 - pt 4 - add shared pref serializer/deserializer to RDD process. r=kmag
#   
#   Depends on D26568
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D26569
# ==============================================================================

diff -r fcab7823cf27 -r dd05584049c6 dom/media/ipc/RDDProcessImpl.cpp
--- a/dom/media/ipc/RDDProcessImpl.cpp	Tue Apr 09 21:07:47 2019 +0000
+++ b/dom/media/ipc/RDDProcessImpl.cpp	Tue Apr 09 21:08:00 2019 +0000
@@ -11,6 +11,8 @@
 #  include "mozilla/sandboxTarget.h"
 #endif
 
+#include "ProcessUtils.h"
+
 namespace mozilla {
 
 using namespace ipc;
@@ -25,15 +27,48 @@
   mozilla::SandboxTarget::Instance()->StartSandbox();
 #endif
   char* parentBuildID = nullptr;
+  char* prefsHandle = nullptr;
+  char* prefMapHandle = nullptr;
+  char* prefsLen = nullptr;
+  char* prefMapSize = nullptr;
   for (int i = 1; i < aArgc; i++) {
     if (!aArgv[i]) {
       continue;
     }
     if (strcmp(aArgv[i], "-parentBuildID") == 0) {
       parentBuildID = aArgv[i + 1];
+
+#ifdef XP_WIN
+    } else if (strcmp(aArgv[i], "-prefsHandle") == 0) {
+      if (++i == aArgc) {
+        return false;
+      }
+      prefsHandle = aArgv[i];
+    } else if (strcmp(aArgv[i], "-prefMapHandle") == 0) {
+      if (++i == aArgc) {
+        return false;
+      }
+      prefMapHandle = aArgv[i];
+#endif
+    } else if (strcmp(aArgv[i], "-prefsLen") == 0) {
+      if (++i == aArgc) {
+        return false;
+      }
+      prefsLen = aArgv[i];
+    } else if (strcmp(aArgv[i], "-prefMapSize") == 0) {
+      if (++i == aArgc) {
+        return false;
+      }
+      prefMapSize = aArgv[i];
     }
   }
 
+  SharedPreferenceDeserializer deserializer;
+  if (!deserializer.DeserializeFromSharedMemory(prefsHandle, prefMapHandle,
+                                                prefsLen, prefMapSize)) {
+    return false;
+  }
+
   return mRDD.Init(ParentPid(), parentBuildID, IOThreadChild::message_loop(),
                    IOThreadChild::channel());
 }
